src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,117 @@
+                this.addIndex(talk, indexValue);
+                maxTalkIndex += 1;
+            }
+        }
+    }
+
+    /**
+     * Returns the greatest index of all log children of an archive node.
+     * @param logs A list of XML objects for individual log nodes (e. g.
+     *  <log id="3" index="1" title="title3"/>
+     * @return Highest value of the index attribute of all log nodes contained
+     *  in the logs list.
+     */
+    private int getMaxLogIndex(final List<XML> logs) {
+        int maxLogIndex = 0;
+        for (final XML log : logs) {
+            int curIndex = 0;
+            final List<String> indexTexts = log.xpath("@index");
+            if (indexTexts.size() == 1) {
+                final String indexText = indexTexts.get(0);
+                try {
+                    curIndex = Integer.parseInt(indexText);
+                } catch (final NumberFormatException exception) {
+                    Logger.error(
+                        this,
+                        String.format(
+                            "Invalid index number '%s'",
+                            indexText
+                        )
+                    );
+                }
+                if (curIndex > maxLogIndex) {
+                    maxLogIndex = curIndex;
+                }
+            }
+        }
+        return maxLogIndex;
+    }
+
+    /**
+     * Returns the highest index of all log nodes in all talks contained in
+     *  the talks list.
+     * @param talks The list of talks to traverse.
+     * @return Highest value of the index attribute of all talks in the talks
+     *  list
+     * @throws IOException Thrown, when problems with reading XML occur.
+     */
+    private int getMaxTalkIndex(final Talks talks) throws IOException {
+        int maxIndex = 0;
+        for (final Talk talk : talks.active()) {
+            final int talkIndex = this.getMaxLogIndex(talk.read()
+                .nodes(ARCHIVE_LOG)
+            );
+            if (talkIndex > maxIndex) {
+                maxIndex = talkIndex;
+            }
+        }
+        return maxIndex;
+    }
+
+    /**
+     * Adds a request tag to a talk node.
+     * @param talk Talk, to which the request node should be added.
+     * @param index Value of the index attribute of the newly created request
+     *  node.
+     * @throws IOException Thrown, when problems with reading XML occur.
+     */
+    private void addIndex(final Talk talk, final int index) throws IOException {
+        talk.modify(
+            new Directives().xpath("//talk").add("request")
+            .attr(INDEX, Integer.toString(index))
+            .attr("id", this.createRequestId())
+            .add("type").set(INDEX)
+            .up()
+            .add("args")
+        );
+    }
+
+    /**
+     * Creates a unique alphanumeric identifier.
+     * @return Random unique identifier without dashes (only numbers and
+     *  letters).
+     */
+    private String createRequestId() {
```

Comment:
Moving this code to a separate method doesn't solve the creation object in a loop issue, it hides the issue.
So, it's better to suppress this warning.